src/foam/lib/json/Outputter.java

```diff
@@ -96,17 +97,33 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
-    writer_.append("\"");
-    writer_.append(escape(s));
-    writer_.append("\"");
+    if ( multiLineOutput_ && s.indexOf('\n') >= 0 ) {
+      writer_.append("\n  \"\"\"\n    ");
+      s = ltrim(s, 4);
```

Comment:
So we're going to trim the 4 space indentation off the string before indentation?  I don't understand why.

Reply:
this is to prevent the indentation from piling up as it's read and outputted; you said this was a problem with adding indentation so it checks to see if it already indented and removes it if needed

Reply:
I don't think this will work the way you are expecting it to work.  You're trimming the String that is being output at output time but instead we should be trimming away the indentation at read time.

If I have an object with a string proprety, and I start that string with four spaces I expect the JSON outputter to preserve those four spaces in its output so that when I read it back those four spaces are still there.  Similarly if I don't have 4 spaces, if I have 0 spaces or 30 spaces at the start of my line, then I expect those to still be there.  It's my data, the outputter shouldn't be modifying my data except in ways that the parser will undo at read time.  I don't think this trim fits that criteria since it always strips up to 4 spaces from my data, but the parser would then have no idea how many spaces to insert at parse time.

Are you sure this setup preserves all indentation in Scripts after being saved and re-read?